parlai/chat_service/services/websocket/agents.py

```diff
@@ -5,29 +5,85 @@
 # LICENSE file in the root directory of this source tree.
 
 import logging
+import base64
+from io import BytesIO
 from parlai.chat_service.core.agents import ChatServiceAgent
+from PIL import Image
 
 
 class WebsocketAgent(ChatServiceAgent):
     """Class for a person that can act in a ParlAI world via websockets"""
 
-    def __init__(self, *args):
-        super().__init__(*args)
+    def __init__(self, opt, manager, receiver_id, task_id):
+        super().__init__(opt, manager, receiver_id, task_id)
         self.message_partners = []
         self.action_id = 1
 
     def observe(self, act):
-        """Send an agent a message through the websocket manager"""
+        """Send an agent a message through the websocket manager
+
+        Only attachments of type `image` are currently supported. In the case of
+        images, the resultant message will have a `text` field which will be a
+        base 64 encoded image and `mime_type` which will be an image mime type.
+
+        Args:
+            act: dict. If act contains an `attachment` key, then a dict should be
+                provided for the value in `attachment`. Otherwise, act should be
+                a dict with the key `text` for the message.
+                For the `attachment` dict, this agent expects a `type` key, which
+                specifies whether or not the attachment is an image. If the
+                attachment is an image, either a `path` key must be specified
+                for the path to the image, or a `data` key holding a PIL Image.
+                A `quick_replies` key can be provided with a list of string quick
+                replies for any message
+        """
         logging.info(f"Sending new message: {act}")
-        if 'payload' in act:
-            raise ValueError("Payload not supported yet by websockets")
+        quick_replies = act.get('quick_replies', None)
```

Comment:
do quick replies make sense in the context of websockets?

Reply:
I included quick_replies in the message structure primarily since some UI's may want to implement a form of quick replies. It can be ignored by the WebSocket client or handled in any way so it's just an extra feature of the messages. It also allows for a more uniform protocol between various chat services so tasks can be agnostic of whether the service supports quick_replies.